File: src/calendar-backend/Calendar.Mongo.Db/Models/Calendar.cs
using MongoDB.Bson.Serialization.Attributes;

namespace Calendar.Mongo.Db.Models
{
    public class Calendar
    {
        [BsonId]
        [BsonRepresentation(MongoDB.Bson.BsonType.ObjectId)]
        public string? Id { get; set; }
        public string? Name { get; set; }   
        public DateTime StartDate { get; set; }
        public List<Lecture>? Lectures { get; set; }
        public DateTimeOffset CreatedDate { get; set; }
    }
}

File: src/calendar-client/src/models/calendar.ts
import { CalendarEvent } from './calendarEvent'
import { Moment } from "moment/moment";

export interface Calendar {
    id?: string
    name: string
    startDate: Moment
    events?: CalendarEvent[]
    lastUpdateDate?: Moment
    createdDate?: Moment
}

export interface KeycloakCalendar {
    keycloakGroupId: string
    name: string
}
